configure.ac

```diff
@@ -497,19 +497,78 @@ if test "x$build_dtls" = "xyes"; then
                   [have_wolfssl="yes"],
                   [have_wolfssl="no"])
 
-    # Mbed TLS [does not have mbedtls.pc pkg-config file]
-    AC_CHECK_LIB(mbedtls, mbedtls_version_get_string,
-                 [have_mbedtls="yes"; MbedTLS_CFLAGS="" ; MbedTLS_LIBS="-lmbedtls -lmbedcrypto -lmbedx509"],
-                 [have_mbedtls="no"], -lmbedx509 -lmbedcrypto)
+    # Mbed TLS
+    if test "x${MbedTLS_CFLAGS+set}" = "xset"; then
+        mbedtls_cflags_overridden="yes"
+    else
+        mbedtls_cflags_overridden="no"
+    fi
+    # Mbed TLS
+    if test "x${MbedTLS_LIBS+set}" = "xset"; then
+        mbedtls_libs_overridden="yes"
+    else
+        mbedtls_libs_overridden="no"
+    fi
+
+    # If MbedTLS_CFLAGS and MbedTLS_LIBS are overridden, pkg-config will always assume the library was found, even
+    # though MbedTLS doesn't necessarily have a pkg-config file. Therefore, we might as well use the "old" way to
+    # determine whether we can link MbedTLS (and which version we have).
+    if test "x$mbedtls_libs_overridden" = "xyes" -a "x$mbedtls_cflags_overridden" = "xyes"; then
```

Comment:
I think that this should be (OR)
````
  if test "x$mbedtls_libs_overridden" = "xyes" -o "x$mbedtls_cflags_overridden" = "xyes"; then
````

Reply:
It was actually intended to be AND.

The issue I was trying to avoid here is the following scenario:
- Both `MbedTLS_CFLAGS` and `MbedTLS_LIBS` are set as environment variables/overridden.
- The MbedTLS version provided through these environment variables does not have a pkg-config file.

In this situation, `PKG_CHECK_MODULES` would simply assume that the library was found without ever executing pkg-config, at least based on my observations. [^1]

Later on, the configure-script would then want to check the version using pkg-config (line 559), and fail because there actually is no pkg-config file to get the version from. 

If either `MbedTLS_CFLAGS` or `MbedTLS_LIBS` are overridden (but not both) this issue cannot occur, as `PKG_CHECK_MODULES` will run pkg-config to determine the missing variable.

[^1]: This behavior is at least hinted at in the [man pages](https://manpages.debian.org/testing/pkg-config/pkg-config.1.en.html#AUTOCONF_MACROS), although it is not outright stated.

Reply:
Testing on Ubuntu 22 confirms this needs to be an AND here.